### PR TITLE
Add Tencent Hunyuan chat model

### PR DESCRIPTION
**Pull request by johnmai-dev**
### Description:
The Tencent Hunyuan model, developed by Tencent, is a large language model by robust Chinese text generation capabilities, adeptness in logical reasoning within complex contexts, and reliable task execution proficiency.For more information, see [https://cloud.tencent.com/document/product/1729](https://cloud.tencent.com/document/product/1729)

### CODE REVIEW

**Comment by eyurtsev** [comment]
@baskaryan this looks good to me, but I don't know if we require integration tests or not. I'd probably like to start requiring usage of SecretKey for all integrations that deal with secrets

File: libs/langchain/langchain/chat_models/hunyuan.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    def lc_serializable(self) -> bool:
+        return True
+
+    hunyuan_api_base: str = "https://hunyuan.cloud.tencent.com"
+    """Hunyuan custom endpoints"""
+    hunyuan_app_id: Optional[str] = None
+    """Hunyuan App ID"""
+    hunyuan_secret_id: Optional[str] = None
+    """Hunyuan Secret ID"""
+    hunyuan_secret_key: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (maiqingqiang) [reply libs/langchain/langchain/chat_models/hunyuan.py:113]: 👌

**Review by eyurtsev — APPROVED** [review]

**Comment by a289459798** [comment]
混元模型报错  Please check app_id parameter type got string expected uint64

**Comment by maiqingqiang** [comment]
> 混元模型报错 Please check app_id parameter type got string expected uint64

已处理，可以更新到v0.0.339

**Pull request merged**